searx/weather.py

```diff
@@ -0,0 +1,607 @@
+import dataclasses
+
+from urllib.parse import quote_plus
+
+import babel
+import babel.numbers
+import babel.dates
+import babel.languages
+
+from searx import network
+from searx.cache import ExpireCache, ExpireCacheCfg
+from searx.extended_types import sxng_request
+from searx.wikidata_units import convert_to_si, convert_from_si
+
+WEATHER_DATA_CACHE: ExpireCache = None  # type: ignore
+"""A simple cache for weather data (geo-locations, icons, ..)"""
+
+YR_WEATHER_SYMBOL_URL = "https://raw.githubusercontent.com/nrkno/yr-weather-symbols/refs/heads/master/symbols/outline"
+
+
+def get_WEATHER_DATA_CACHE():
+
+    global WEATHER_DATA_CACHE  # pylint: disable=global-statement
+
+    if WEATHER_DATA_CACHE is None:
+        WEATHER_DATA_CACHE = ExpireCache.build_cache(
+            ExpireCacheCfg(
+                name="WEATHER_DATA_CACHE",
+                MAX_VALUE_LEN=1024 * 200,  # max. 200kB per icon (icons have most often 10-20kB)
+                MAXHOLD_TIME=60 * 60 * 24 * 7 * 4,  # 4 weeks
+            )
+        )
+    return WEATHER_DATA_CACHE
+
+
+def _get_sxng_locale_tag() -> str:
+    # The function should return a locale (the sxng-tag: de-DE.en-US, ..) that
+    # can later be used to format and convert measured values for the output of
+    # weather data to the user.
+    #
+    # In principle, SearXNG only has two possible parameters for determining
+    # the locale: the UI language or the search- language/region.  Since the
+    # conversion of weather data and time information is usually
+    # region-specific, the UI language is not suitable.
+    #
+    # It would probably be ideal to use the user's geolocation, but this will
+    # probably never be available in SearXNG (privacy critical).
+    #
+    # Therefore, as long as no "better" parameters are available, this function
+    # returns a locale based on the search region.
+
+    # pylint: disable=import-outside-toplevel,disable=cyclic-import
+    from searx import query
+    from searx.preferences import ClientPref
+
+    query = query.RawTextQuery(sxng_request.form.get("q", ""), [])
+    if query.languages and query.languages[0] not in ["all", "auto"]:
+        return query.languages[0]
+
+    search_lang = sxng_request.form.get("language")
+    if search_lang and search_lang not in ["all", "auto"]:
+        return search_lang
+
+    client_pref = ClientPref.from_http_request(sxng_request)
+    search_lang = client_pref.locale_tag
+    if search_lang and search_lang not in ["all", "auto"]:
+        return search_lang
+    return "en"
+
+
+def symbol_url(condition: WeatherConditionType) -> str | None:
+    """Returns ``data:`` URL for the weather condition symbol or ``None`` if
+    the condition is not of type :py:obj:`WeatherConditionType`.
+
+    If symbol (SVG) is not already in the :py:obj:`WEATHER_DATA_CACHE` its
+    fetched from https://github.com/nrkno/yr-weather-symbols
+    """
+    # Symbols for darkmode/lightmode? .. and day/night symnbols? .. for latter
```

Comment:
```suggestion
    # Symbols for darkmode/lightmode? .. and day/night symnbols? .. for later
```

Reply:
Its not the word "later" in sense of *time* ... its the word "latter" in sense of *the latest* in the list given before